### DISCUSSION TITLE
FrontEnd Implementation: Choose a Framework

### DISCUSSION THREAD

**Discussion by Ar2691** [uprm-inso4101-2024-2025-s1/semester-project-regiupr#5] [General]
For web development what framework for the frontend should we use?

### Options
 
**Streamlit**: Streamlit is particularly suited for those who want to create data-driven web applications quickly 
without delving deeply into web development. It provides a straightforward way to build interactive interfaces 
using Python.

**Angular**: Angular is a comprehensive front-end web application framework. It’s designed to build dynamic, 
single-page applications (SPAs) and offers a robust suite of features for managing application complexity.

**React**: React is a popular JavaScript library for building user interfaces, particularly single-page applications
where you want to have a dynamic user experience.

**Vue**: Vue is a progressive JavaScript framework used for building user interfaces. Unlike monolithic frameworks, 
Vue is designed from the ground up to be incrementally adoptable, meaning you can use as much or as little of it
as you need.

These are a few options I looked into, feel free to add more in the comments or pick one.

**Comment by cristianMarcial** [comment]
Aren't libraries like React, Vue and others specifically intended to support javascript programs for use in web browsers? I don't understand why we would use them to create an executable app outside a web browser. In any case, Python already has standard libraries for creating interactive graphical interfaces for applications.

**Comment by Rezanmonis** [comment]
We are looking for a frontend developer as our teammate for our edtech project rezanmonis@gmail.com

**Comment by Yamil-Serrano** [comment]
Although Python has dedicated libraries for creating graphical interfaces, from the little research I have been able to do, Steamlit seems like a good option for reading data and creating interfaces.

**Comment by nicolle-feliciano** [comment]
I agree with Yamil, Streamlit sounds like a good framework to use.

**Comment by J3SSY-ANDU** [comment]
None of the libraries mentioned—Streamlit, Angular, React, or Vue—are designed for building true native desktop applications. These are all web development frameworks primarily intended for creating web interfaces and cannot provide the full functionality or performance of a native desktop app. For desktop development, consider using libraries like PyQt or Tkinter for Python, and Electron or Flutter for JavaScript.
